### PR TITLE
Less verbose MockedConsumer by default

### PR DESCRIPTION
**Pull request by zupo**
But keep the option to turn up the verbosity if needed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ If you want to skip sending some `track` or `profile_set` calls to Customer.io,
 - Provides dataclasses for events and properties, to avoid typos.
 - You can roll your own [`Consumer`](https://mixpanel.github.io/mixpanel-python/#built-in-consumers), for example one that schedules a background task to send events, to increase request processing speed, since HTTP requests to Mixpanel are offloaded to a background task.
 - Provides a MixpanelQuery helper to use [JQL](https://mixpanel.com/jql/) to query Mixpanel for data. Some common queries like one for getting profiles by email are included.
-- In local development and unit testing, all messages are stored in `request.mixpanel.api._consumer.mocked_messages` which makes writing integration tests a breeze.
+- In local development and unit testing, all messages are stored in `request.mixpanel.api._consumer.mocked_messages` which makes writing integration tests a breeze. By default, these messages omit "system" properties such as `token`, `time`, `mp_lib` and similar, to make tests less verbose. If you need them, set `MockedConsumer.DROP_SYSTEM_MESSAGE_PROPERTIES` to `True`.
```
Comment (am-on) [inline README.md]:
```suggestion
- In local development and unit testing, all messages are stored in `request.mixpanel.api._consumer.mocked_messages` which makes writing integration tests a breeze. By default, these messages omit "system" properties such as `token`, `time`, `mp_lib` and similar, to make tests less verbose. If you need them, set `MockedConsumer.DROP_SYSTEM_MESSAGE_PROPERTIES` to `False`.
```

Comment (zupo) [reply README.md]: 🤦 

Thanks for the catch!

**Review by am-on — APPROVED** [review]

**Comment by zupo** [comment]
Which test case do you prefer? Why?

A) plain dict

```python
    assert m.api._consumer.mocked_messages == [
        {
            "endpoint": "people",
            "msg": {
                "$distinct_id": "foo",
                "$set": {"$created": "2020-02-02T01:01:00"},
            },
        },
        {
            "endpoint": "customer.io",
            "msg": {
                "id": "foo",
                "created_at": 1580605260,
            },
        },
    ]
```

B) bespoke class

```python
    assert m.api._consumer.mocked_messages == [
        MockedMessage(
            endpoint="people",
            msg={
                "$distinct_id": "foo",
                "$set": {"$created": "2020-02-02T01:01:00"},
            },
        ),
        MockedMessage(
            endpoint="customer.io",
            msg={
                "id": "foo",
                "created_at": 1580605260,
            },
        ),
    ]
```

**Comment by zupo** [comment]
pytest diff for a failed assert for B)

Screenshot 2022-03-21 at 16 03 28

**Comment by zupo** [comment]
pytest diff for a failed assert for A)

Screenshot 2022-03-21 at 16 05 35

**Comment by zupo** [comment]
Which test case do you prefer? Why?

See examples in comments above.

* diffs are almost identical, no clear winners there
* both can be copy/pasted directly from PDB into test case code and auto-formatted with Black
* FOR MockedMessages: plain dicts are harder to "scan" with eyes, as everything is text, syntax highlighting doesn't assist you
* FOR MockedMessage: closer to how Mixpanel really works, so less "abstraction fog"
* AGAINST MockedMessages: need to remember to import MockedMessage class in every test case

@dz0ny, @am-on, @karantan, @myrheimb: please vote 🚀  for plain dict and 🎉   for `MockedMessage`. 😕  for "I defer the decision to @zupo".

**Comment by dz0ny** [comment]
assert via dict gives you direct result no need for debugger.

**Comment by zupo** [comment]
> assert via dict gives you direct result no need for debugger.

Both approaches have dicts. A) is full dict, B) is a class where the message is a dict. I showed above the results in assert failures are identical. What am I missing?

**Comment by am-on** [comment]
I prefer dict. If I don't see any previous test cases I might think I have to do 
```python
assert m.api._consumer.mocked_messages[0].endpoint == "customer.io"
assert m.api._consumer.mocked_messages[0].msg == {
                "id": "foo",
                "created_at": 1580605260,
            },
```

instead of 
```python
m.api._consumer.mocked_messages[0] == MockedMessage(
            endpoint="customer.io",
            msg={
                "id": "foo",
                "created_at": 1580605260,
            },
        ),
```

**Comment by dz0ny** [comment]
> > assert via dict gives you direct result no need for debugger.
> 
> Both approaches have dicts. A) is full dict, B) is a class where the message is a dict. I showed above the results in assert failures are identical. What am I missing?

MockedMessage has hidden content unless you unwrap it in debugger.

**Comment by zupo** [comment]
> > > assert via dict gives you direct result no need for debugger.
> > 
> > 
> > Both approaches have dicts. A) is full dict, B) is a class where the message is a dict. I showed above the results in assert failures are identical. What am I missing?
> 
> MockedMessage has hidden content unless you unwrap it in debugger.

No, it doesn't, this is how it looks:

``` python
(Pdb) m.api._consumer.mocked_messages
[MockedMessage(endpoint='people', msg={'$distinct_id': 'foo', '$set': {'$created': '2020-02-02T01:01:00'}}), {'endpoint': 'customer.io', 'msg': {'id': 'foo', 'created_at': 1580605260}}]
```

Or did you have sth. else in mind?

**Comment by zupo** [comment]
> I prefer dict. If I don't see any previous test cases I might think I have to do
> 
> ```python
> assert m.api._consumer.mocked_messages[0].endpoint == "customer.io"
> assert m.api._consumer.mocked_messages[0].msg == {
>                 "id": "foo",
>                 "created_at": 1580605260,
>             },
> ```
> 
> instead of
> 
> ```python
> m.api._consumer.mocked_messages[0] == MockedMessage(
>             endpoint="customer.io",
>             msg={
>                 "id": "foo",
>                 "created_at": 1580605260,
>             },
>         ),
> ```

Any idea what makes you think like this? 

Could it be that this is recency bias? I.e. all asserts have been written like this *up to this point*. Going forward, if this gets merged, I'll rewrite all asserts to be according to this new format.

**Comment by dz0ny** [comment]
> > > > assert via dict gives you direct result no need for debugger.
> > > 
> > > 
> > > Both approaches have dicts. A) is full dict, B) is a class where the message is a dict. I showed above the results in assert failures are identical. What am I missing?
> > 
> > 
> > MockedMessage has hidden content unless you unwrap it in debugger.
> 
> No, it doesn't, this is how it looks:
> 
> ```python
> (Pdb) m.api._consumer.mocked_messages
> [MockedMessage(endpoint='people', msg={'$distinct_id': 'foo', '$set': {'$created': '2020-02-02T01:01:00'}}), {'endpoint': 'customer.io', 'msg': {'id': 'foo', 'created_at': 1580605260}}]
> ```
> 
> Or did you have sth. else in mind?

I have used pdb twice :D and I'm pretty sure the output was <instance of MockedMessage> or something like that, never the full dump of a class. So that's new to me.

**Comment by zupo** [comment]
> I have used pdb twice :D

😱 

How .... what ... huh?? How do you debug then? :) I.e. what tooling do you use, so I can test what the difference is between A) and B)?

**Comment by am-on** [comment]
>Any idea what makes you think like this?
>
> Could it be that this is recency bias? I.e. all asserts have been written like this up to this point. Going forward, if this gets merged, I'll rewrite all asserts to be according to this new format.

It's more like "oh it's a class":
- can I compare two instances? Does it have implemented `__eq__`?
- from where do I import it?
- can I create a new instance that can be comparable - is there some data that is different for each instance?

I have to go over all this or do a quick and ugly one with:
```python
assert m.api._consumer.mocked_messages[0].endpoint == "customer.io"
```

**Comment by zupo** [comment]
Let's go with plain dict then.

**Review by dz0ny — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ If you want to skip sending some `track` or `profile_set` calls to Customer.io,
 - Provides dataclasses for events and properties, to avoid typos.
 - You can roll your own [`Consumer`](https://mixpanel.github.io/mixpanel-python/#built-in-consumers), for example one that schedules a background task to send events, to increase request processing speed, since HTTP requests to Mixpanel are offloaded to a background task.
 - Provides a MixpanelQuery helper to use [JQL](https://mixpanel.com/jql/) to query Mixpanel for data. Some common queries like one for getting profiles by email are included.
-- In local development and unit testing, all messages are stored in `request.mixpanel.api._consumer.mocked_messages` which makes writing integration tests a breeze.
+- In local development and unit testing, all messages are stored as plain dicts in `request.mixpanel.api._consumer.mocked_messages`. This makes writing integration tests a breeze. By default, these dicts omit "system" properties such as `token`, `time`, `mp_lib` and similar, to make tests less verbose. If you need them, set `MockedConsumer.DROP_SYSTEM_MESSAGE_PROPERTIES` to `False`.
```
Comment (dz0ny) [inline README.md]:
```suggestion
- In local development and unit testing, all messages are stored as plain dicts in `request.mixpanel.api._consumer.mocked_messages`. This makes writing integration tests a breeze. By default, these dicts omit "library" properties such as `token`, `time`, `mp_lib` and similar, to make tests less verbose. If you need them, set `MockedConsumer.DROP_SYSTEM_MESSAGE_PROPERTIES` to `False`.
```

Comment (zupo) [reply README.md]: Tnx!

**Comment by zupo** [comment]
Realized another improvement is possible: instead of `mixpanel.api._consumer.mocked_messages` we can shortcut it to `mixpanel.mocked_messages`.

**Comment by zupo** [comment]
I've rewritten over 100 tests in [Pareto Dash](https://paretosecurity.com/) over to this branch, and they look far better. Ready to merge!

**Pull request merged**